### PR TITLE
feat: update permission & permission rule when role deleted

### PR DESCRIPTION
**Pull request by notdu**

### CODE REVIEW

**Comment by notdu** [comment]
hi @hsluoyz, did you go to sleep? :v

**Comment by hsluoyz** [comment]
@notdu it's daylight now. What's this PR for?

**Comment by notdu** [comment]
When a role is deleted, all the permissions contain that role should be updated as well. 
All the related permission rules should be deleted too.

File: object/role.go [hunk 1]
```diff
@@ -133,11 +133,29 @@ func AddRole(role *Role) bool {
...
 
 func DeleteRole(role *Role) bool {
+	roleId := role.GetId()
+	permissions := GetPermissionsByRole(roleId)
+	for _, permission := range permissions {
+		removeGroupingPolicies(permission)
+		removePolicies(permission)
+
+		permission.Roles = util.DeleteVal(permission.Roles, roleId)
+		_, err := adapter.Engine.ID(core.PK{permission.Owner, permission.Name}).AllCols().Update(permission)
```
Comment (hsluoyz) [inline object/role.go:143]: use `updatePermission` (create it it not exist)

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**